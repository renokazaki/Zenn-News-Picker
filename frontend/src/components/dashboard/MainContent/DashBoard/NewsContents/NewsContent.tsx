import { CardContent } from "../../../../ui/card";
import type { NewsItem } from "../../../../../type/News.type";

interface NewsContentProps {
  news: NewsItem;
}

const NewsContent = ({ news }: NewsContentProps) => {
  return (
    <CardContent className="pt-0 pb-4">
      <p className="text-sm text-muted-foreground line-clamp-10">{news.text}</p>
    </CardContent>
  );
};

export default NewsContent;
