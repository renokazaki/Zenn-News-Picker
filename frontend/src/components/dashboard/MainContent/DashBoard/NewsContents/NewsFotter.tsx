import { CardFooter } from "../../../../ui/card";
import { Clock, ExternalLink } from "lucide-react";
import type { NewsItem } from "../../../../../type/News.type";
import { formatDate } from "../../../../../lib/formatdate";

interface NewsFotterProps {
  news: NewsItem;
}

const NewsFotter = ({ news }: NewsFotterProps) => {
  return (
    <CardFooter className="pt-0 flex items-center justify-between">
      <div className="flex items-center gap-1 text-xs text-muted-foreground">
        <Clock className="h-3 w-3" />
        <time dateTime={news.publishedAt}>{formatDate(news.publishedAt)}</time>
      </div>
      <ExternalLink className="h-3 w-3 text-muted-foreground group-hover:text-primary transition-colors" />
    </CardFooter>
  );
};

export default NewsFotter;
