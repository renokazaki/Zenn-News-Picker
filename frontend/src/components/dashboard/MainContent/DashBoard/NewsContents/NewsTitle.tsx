import { CardHeader } from "../../../../ui/card";
import type { NewsItem } from "../../../../../type/News.type";

interface NewsTitleProps {
  news: NewsItem;
}

const NewsTitle = ({ news }: NewsTitleProps) => {
  return (
    <CardHeader className="">
      <h3 className="font-semibold text-sm leading-tight line-clamp-3 group-hover:text-primary transition-colors">
        {news.title}
      </h3>
    </CardHeader>
  );
};

export default NewsTitle;
