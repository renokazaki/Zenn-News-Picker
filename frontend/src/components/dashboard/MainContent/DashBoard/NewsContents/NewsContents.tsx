import { Card } from "../../../../../components/ui/card";
import { useNews } from "../useNews.Hook";
import type { NewsItem, DateType } from "../../../../../type/News.type";
import NewsTitle from "./NewsTitle";
import NewsContent from "./NewsContent";
import NewsFotter from "./NewsFotter";

interface NewsContentsProps {
  selectedDate?: DateType;
}

const NewsContents = ({ selectedDate }: NewsContentsProps) => {
  const news: NewsItem[] = useNews(selectedDate);

  if (news.length === 0 && selectedDate) {
    return (
      <div className="col-span-full flex justify-center items-center py-8">
        <p className="text-muted-foreground">
          {selectedDate.toLocaleDateString("ja-JP")}のニュースはありません
        </p>
      </div>
    );
  }

  return (
    <>
      {news.map((news, id) => (
        <Card
          key={id}
          className="group hover:shadow-lg transition-all duration-200 hover:-translate-y-1 cursor-pointer border-0 shadow-md bg-gradient-to-br from-white to-gray-50/50 dark:from-gray-900 dark:to-gray-800/50"
          onClick={() => window.open(news.url, "_blank")}
        >
          <NewsTitle news={news} />
          <NewsContent news={news} />
          <NewsFotter news={news} />
        </Card>
      ))}
    </>
  );
};

export default NewsContents;
