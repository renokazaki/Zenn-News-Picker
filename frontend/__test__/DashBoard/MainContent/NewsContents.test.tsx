import { render, screen } from "@testing-library/react";
import React from "react";
import NewsContents from "../../../src/components/dashboard/MainContent/DashBoard/NewsContents/NewsContents";

// useNews.Hook.ts のモック
jest.mock(
  "../../../src/components/dashboard/MainContent/DashBoard/useNews.Hook",
  () => ({
    useNews: jest.fn().mockReturnValue([
      {
        title: "テストニュース1",
        url: "https://example.com/1",
        text: "これはテストニュースです",
        publishedAt: "2025-09-10T10:00:00Z",
      },
      {
        title: "テストニュース2",
        url: "https://example.com/2",
        text: "これは2つ目のテストニュースです",
        publishedAt: "2025-09-11T15:30:00Z",
      },
    ]),
  })
);

describe("NewsContents", () => {
  let container: HTMLElement;

  beforeEach(() => {
    const renderResult = render(<NewsContents selectedDate={new Date()} />);
    container = renderResult.container;
  });

  test("正しくレンダリングされること", () => {
    expect(container).toBeDefined();
    expect(screen.getByText("テストニュース1")).toBeDefined();
    expect(screen.getByText("テストニュース2")).toBeDefined();
    expect(screen.getByText("これはテストニュースです")).toBeDefined();
    expect(screen.getByText("これは2つ目のテストニュースです")).toBeDefined();
    expect(screen.getByText("2025-09-10")).toBeDefined();
    expect(screen.getByText("2025-09-11")).toBeDefined();
  });

  test("スナップショットと一致すること", () => {
    expect(container).toMatchSnapshot();
  });
});
