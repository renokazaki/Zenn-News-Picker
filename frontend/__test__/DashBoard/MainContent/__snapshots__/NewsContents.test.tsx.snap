// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`NewsContents スナップショットと一致すること 1`] = `
<div>
  <div
    class="text-card-foreground flex flex-col gap-6 rounded-xl py-6 group hover:shadow-lg transition-all duration-200 hover:-translate-y-1 cursor-pointer border-0 shadow-md bg-gradient-to-br from-white to-gray-50/50 dark:from-gray-900 dark:to-gray-800/50"
    data-slot="card"
  >
    <div
      class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-1.5 px-6 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6"
      data-slot="card-header"
    >
      <h3
        class="font-semibold text-sm leading-tight line-clamp-3 group-hover:text-primary transition-colors"
      >
        テストニュース1
      </h3>
    </div>
    <div
      class="px-6 pt-0 pb-4"
      data-slot="card-content"
    >
      <p
        class="text-sm text-muted-foreground line-clamp-10"
      >
        これはテストニュースです
      </p>
    </div>
    <div
      class="px-6 [.border-t]:pt-6 pt-0 flex items-center justify-between"
      data-slot="card-footer"
    >
      <div
        class="flex items-center gap-1 text-xs text-muted-foreground"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-clock h-3 w-3"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 6v6l4 2"
          />
          <circle
            cx="12"
            cy="12"
            r="10"
          />
        </svg>
        <time
          datetime="2025-09-10T10:00:00Z"
        >
          2025-09-10
        </time>
      </div>
      <svg
        aria-hidden="true"
        class="lucide lucide-external-link h-3 w-3 text-muted-foreground group-hover:text-primary transition-colors"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15 3h6v6"
        />
        <path
          d="M10 14 21 3"
        />
        <path
          d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
        />
      </svg>
    </div>
  </div>
  <div
    class="text-card-foreground flex flex-col gap-6 rounded-xl py-6 group hover:shadow-lg transition-all duration-200 hover:-translate-y-1 cursor-pointer border-0 shadow-md bg-gradient-to-br from-white to-gray-50/50 dark:from-gray-900 dark:to-gray-800/50"
    data-slot="card"
  >
    <div
      class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-1.5 px-6 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6"
      data-slot="card-header"
    >
      <h3
        class="font-semibold text-sm leading-tight line-clamp-3 group-hover:text-primary transition-colors"
      >
        テストニュース2
      </h3>
    </div>
    <div
      class="px-6 pt-0 pb-4"
      data-slot="card-content"
    >
      <p
        class="text-sm text-muted-foreground line-clamp-10"
      >
        これは2つ目のテストニュースです
      </p>
    </div>
    <div
      class="px-6 [.border-t]:pt-6 pt-0 flex items-center justify-between"
      data-slot="card-footer"
    >
      <div
        class="flex items-center gap-1 text-xs text-muted-foreground"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-clock h-3 w-3"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 6v6l4 2"
          />
          <circle
            cx="12"
            cy="12"
            r="10"
          />
        </svg>
        <time
          datetime="2025-09-11T15:30:00Z"
        >
          2025-09-11
        </time>
      </div>
      <svg
        aria-hidden="true"
        class="lucide lucide-external-link h-3 w-3 text-muted-foreground group-hover:text-primary transition-colors"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15 3h6v6"
        />
        <path
          d="M10 14 21 3"
        />
        <path
          d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
        />
      </svg>
    </div>
  </div>
</div>
`;
